src/Symfony/Bundle/FrameworkBundle/Command/SecretsAddCommand.php

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Console command to add an encrypted secret
+ * Usage: php bin/console secrets:add
+ */
+class SecretsAddCommand extends Command
+{
+    protected static $defaultName = 'secrets:add';
+    private $io;
+    private $secretsProcessor;
+
+    public function __construct(SecretVarProcessor $secretsProcessor)
+    {
+        $this->secretsProcessor = $secretsProcessor;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Encrypt and add a secret key value pair to the specified secrets file')
+            ->setHelp(<<<'HELP'
+The <info>%command.name%</info> command adds an encrypted secret to the specified secrets-file
+
+If encrypted_secrets.enabled is set to true in your yaml config,
+
+  <info>php %command.full_name% DATABASE_URL mysql://db_user:db_password@127.0.0.1:3306/db_name</info>
+
+will encrypt the value for DATABASE_URL using the public key file specified in encrypted_secrets.public_key_file, and
+add the pair to the json secrets file configured by encrypted_secrets.secrets_file. If you haven't configured encrypted_secrets or wish to override these values,
+you can provide the required information via the public-key-file and secrets-file options:
+
+<info>php %command.full_name% DATABASE_URL mysql://db_user:db_password@127.0.0.1:3306/db_name --public-key-file ./config/secrets.pub --secrets-file ./config/secrets.jwe</info>
+
+If this is the first time you've run <info>%command.name%</info> and no secrets-file exists in the specified location, one will be created for you.
+
+Always store your private key in a secure location outside of version control; you will not be able to recover your secrets without it.
+HELP
+            )
+            ->addArgument('secret-name', InputArgument::REQUIRED, 'The variable name of the secret (e.g., DATABASE_URL).')
+            ->addArgument('secret-value', InputArgument::REQUIRED, 'The secret to be encrypted.')
+            ->addOption('public-key-file', 'p', InputOption::VALUE_REQUIRED, 'The public key file to be used for encryption.')
+            ->addOption('secrets-file', 's', InputOption::VALUE_REQUIRED, 'The file to write encrypted secrets to')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $name = $input->getArgument('secret-name');
+        $secretValue = $input->getArgument('secret-value');
+        $pubKeyLocation = $input->getOption('public-key-file');
+        $secretsLocation = $input->getOption('secrets-file');
+
+        if (isset($secretsLocation) && !file_exists($secretsLocation)) {
+            JweHandler::initSecretsFile($secretsLocation);
+        }
```

Comment:
What if I don't pass anything to the `--secrets-file` option? Shouldn't the secrets file path (that's in your config) be initialized?

Reply:
yeah with the current logic i believe it would actually blow up on compile if you had it set to something that wasn't there. once I implement the "configure commands through encrypted_secrets config only" logic, (if we still want things to break on compile) i think the flow would be more like-

1) dev configures encrypted_secrets file locations if they require something other than default
2) dev runs a `secrets:enable` command that:
    a) creates all of these files and a keypair if absent and ensures they are writable
    b) sets enabled to true

How does that sound?